core/ui/components/drive_card.gd

```diff
@@ -0,0 +1,180 @@
+extends MarginContainer
+
+var steam_disks := load("res://core/systems/disks/steam_removable_media_manager.tres") as SteamRemovableMediaManager
+var notification_manager := load("res://core/global/notification_manager.tres") as NotificationManager
+
+const partition_card_scene: PackedScene = preload("res://core/ui/components/partition_card.tscn")
+const hdd_icon = preload("res://assets/icons/interface-hdd.svg")
+const nvme_icon = preload("res://assets/icons/interface-nvme.svg")
+const sd_icon= preload("res://assets/icons/interface-sd.svg")
+const ssd_icon= preload("res://assets/icons/interface-ssd.svg")
+const usb_icon= preload("res://assets/icons/interface-usb.svg")
+
+var device: BlockDevice
+var device_path: String
+var highlight_tween: Tween
+
+signal format_drive(device: BlockDevice)
+signal format_sd_card
+signal init_partition(device: PartitionDevice)
+
+@onready var drive_name_label: Label = $%DriveName
+@onready var drive_size_label: Label = $%DriveSize
+@onready var drive_icon: TextureRect = $%IconTextureRect
+@onready var drive_focus_group: FocusGroup = $%DriveFocusGroup
+@onready var format_button: CardButton = $%FormatButton
+@onready var partitions_container: VBoxContainer = $%PartitionsVBox
+@onready var partitions_focus_group: FocusGroup = $%PartitionsFocusGroup
+
+var logger: Logger
+var log_level:= Log.LEVEL.INFO
+
+
+## Performs _ready fucntionality with the given BlockDevice
+func setup(device: BlockDevice) -> void:
+	self.device = device
+	self.device_path = "/dev" + self.device.dbus_path.trim_prefix(steam_disks.BLOCK_PREFIX)
+	logger = Log.get_logger("DriveCard|"+self.device_path, log_level)
+
+	logger.debug("Setup Drive Card")
+	drive_name_label.text = self.device_path
+	drive_size_label.text = _get_readable_size(self.device.block_size)
+
+	format_button.visible = steam_disks.format_capable
+	format_button.pressed.connect(_on_format_drive)
```

Comment:
We should _really_ have a confirmation dialog before doing such a destructive operation.